# Unit 3 Drills

## Lesson 1 Drill

import math
import numpy as np
import statistics as stats

### mean, median, and mode of the kids' ages; variance, standard deviation, and standard error

In [9]:
ages = [14, 12, 11, 10, 8, 6, 8]

In [16]:
n = len(ages)

In [14]:
mean = np.mean(ages)
print(mean)

9.85714285714


In [11]:
np.median(ages)

10.0

In [12]:
stats.mode(ages)

8

In [18]:
# variance (sample)
var = sum((ages - mean) **2) / (n - 1)
print(var)

7.47619047619


In [25]:
# standard deviation (sample)
sd = np.std(ages, ddof=1)
print(sd)
# math formula
if sd != math.sqrt(var): print("ALERT: ERROR IN CALCULATION")

2.73426232761


In [28]:
# standard error of the mean
se = np.std(ages, ddof=1) / np.sqrt(n)
print(se)
# math formula
if se != sd / math.sqrt(n): print("ALERT: ERROR IN CALCULATION")

1.03345401972


### Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?
The mean serves well for central tendency; the skew is slight, and outliers do not appear. The mode 

The small SE of 1.03 highlights the precision of this sample mean estimate.

#### Cindy turns 7: update array

In [33]:
ages2 = [14, 12, 11, 10, 8, 6+1, 8]

mean = np.mean(ages2)

np.median(ages2)

stats.mode(ages2)

# variance (sample)
var = sum((ages2 - mean) **2) / (n - 1)

# standard deviation (sample)
sd = np.std(ages2, ddof=1)
# math formula
if sd != math.sqrt(var): print("ALERT: ERROR IN CALCULATION")

# standard error of the mean
se = np.std(ages2, ddof=1) / np.sqrt(n)
# math formula
if se != sd / math.sqrt(n): print("ALERT: ERROR IN CALCULATION")


The mean changes; median and mode remain the same.

Consequently the variance, standard deviation and standard error all change.

In [45]:
ages3 = [14, 12, 11, 10, 8, 6, 1]  # Cousin Oliver removed

mean = np.mean(ages3)
print("mean is {}".format(mean))

print("median is {}".format(np.median(ages3)))
print("No unique mode. All are unique")

# variance (sample)
var = sum((ages3 - mean) **2) / (n - 1)
print("var is {}".format(var))

# standard deviation (sample)
sd = np.std(ages3, ddof=1)
print("sd is {}".format(sd))

# standard error of the mean
se = np.std(ages3, ddof=1) / np.sqrt(n)
print("se is {}".format(se))


mean is 8.857142857142858
median is 10.0
No unique mode. All are unique
var is 18.80952380952381
sd is 4.3369947901195145
se is 1.6392299502912862


The mean is pulled down precisely one whole year, on account of this one outlier.
The median is unaffected by only one outlier, becoming the most precise indicator of central tendency.
No single mode remains; all ages are unique.

VAR, SD and SE all jump up tremendously.

### Question 5
On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

# Unit 3 Drills (cont.)

## Lesson 2.4 Drill: Probability

#### Calculate the probability of flipping a balanced coin four times and getting each pattern: HTTH, HHHH and TTHH.
(1/2) **4 = 1/16 for each of these ordered patterns.


#### If a list of people has 24 women and 21 men, then the probability of choosing a man from the list is 21/45. What is the probability of not choosing a man?
24/45


#### The probability that Bernice will travel by plane sometime in the next year is 10%. The probability of a plane crash at any time is .005%. What is the probability that Bernice will be in a plane crash sometime in the next year?
.0005%


#### A data scientist wants to study the behavior of users on the company website. Each time a user clicks on a link on the website, there is a 5% chance that the user will be asked to complete a short survey about their behavior on the website. The data scientist uses the survey data to conclude that, on average, users spend 15 minutes surfing the company website before moving on to other things. What is wrong with this conclusion?
The survey only collects data from people who not only opt-in, or self-select, in to the survey, but people who are patient and persistant enough to complete it. One example of website user data not captured are from those with quicker online behavior, a shorter attention span, or less interest in volunteering their time to assist with the survey.

# Unit 3 Drills (cont.)

## Lesson 2 Project 6 Drill: Bayes' Rule

## Lesson 2 Project 7 Drill: Evaluating data sources

Possible shortcomings in terms of bias, objective quality or exceptional circumstance of the data sources
What could be done to either adjust the analysis or reframe the question for accuracy?

Data Source: Amsterdam availability data scraped from AirBnB on December 24th. 
Question: What are the popular neighborhoods in Amsterdam?

    Concern: a major holiday weekend can skew the results. Conclusions can be formed for holiday travel, or for a vacation destination like Amsterdam, but not for an overall general conclusion.

Data Source: Mental health services use on September 12, 2001 in San Francisco, CA and New York City, NY. 
    Question: How do patterns of mental health service use vary between cities?

    Concern: SF and especially NYC were heavily affected by the tragedies on 9/11.  Furthermore, social services including mental health services could very well be under strain in that aftermath.
        
Data Source: Armenian Pub Survey. Question: What are the most common reasons Armenians visit local pubs?

    Concern: This data will only capture info from the Armenians who visit pubs, not the general public, including children or those with temperate preferences on alcohol.  Furthermore, if it's not geographically confined to Armenia, e.g. Armenian pubs in America, then it could capture American customers, whether or not they know it's an Armenian pub.

## Unit 3 Lesson 2 Project 8
### Challenge - Beware of Monty Hall

The idea behind understanding the Monty Hall Problem is applying conditional probability against intuitive feeling.

Suppose you choose door A and then door B is opened, revealing nothing. Door B becomes the condition, the given event.  This means that, given that door B is opened, we calculate the probability of door A having the car, and then door C.

Case 1: player selects door A with car.
Probability: 1/3.
	Case 1a: player selects door A with car, and then Monty opens goat behind B.
	Probability: 1/6.
	Case 1b: player selects door A with car, and then Monty opens goat behind C.
	Probability: 1/6.

Case 2: player selects door A with goat (car behind C).
Probability: 1/3.
	Case 2a: player selects door A with goat, and then Monty opens goat behind B.
	Probability: 1/3.
	Case 2b: player selects door A with goat, and then Monty opens car behind C.
	Probability: 0.

Case 3: player selects door A with goat (car behind B).
Probability: 1/3.
	Case 3a: player selects door A with goat, and then Monty opens goat behind B.
	Probability: 0.
	Case 3b: player selects door A with goat, and then Monty opens car behind C.
	Probability: 1/3.

The car cannot be behind door B since it is opened.  So the possibilities are: Case 1a and 2a. Outside of these possibilities are cases that CANNOT occur (2b, 3a) or DID not occur (of C opening, cases 1b and 3b).